godot-macros/src/class/data_models/group_export.rs

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::class::data_models::fields::Fields;
+use crate::util::KvParser;
+use std::cmp::Ordering;
+
+/// Points to index of a given group name in [Fields.groups](field@Fields::groups).
+///
+/// Two fields with the same GroupIdentifier belong to the same group.
+pub type GroupIdentifier = usize;
+
+pub struct FieldGroup {
+    pub group_name_index: Option<GroupIdentifier>,
+    pub subgroup_name_index: Option<GroupIdentifier>,
+}
+
+impl FieldGroup {
+    fn parse_group(
+        expr: &'static str,
+        parser: &mut KvParser,
+        groups: &mut Vec<String>,
+    ) -> Option<GroupIdentifier> {
+        let group = parser.handle_expr(expr).unwrap_or(None)?.to_string();
+
+        if let Some(group_index) = groups
+            .iter()
+            .position(|existing_group| existing_group == &group)
+        {
+            Some(group_index)
+        } else {
+            groups.push(group);
+            Some(groups.len() - 1)
+        }
+    }
+
+    pub(crate) fn new_from_kv(parser: &mut KvParser, groups: &mut Vec<String>) -> Self {
+        Self {
+            group_name_index: Self::parse_group("group", parser, groups),
+            subgroup_name_index: Self::parse_group("subgroup", parser, groups),
+        }
+    }
+}
```

Comment:
This currently accepts any token-tree and just prays that it's a string litteral. This is also accepted:
```rs
    #[export(group = f64::MAX)]
```

but is turned into a group `"f64 :: MAX"` which is not something I'd expect.

Reply:
fix'd, I also included key with a proper span pointing at the attribute, for example:

```rs
error: missing value for 'subgroup' (must be String literal)
  --> src/lib.rs:27:14
   |
27 |     #[export(subgroup = some subgroup 2)]
   |              ^^^^^^^^

error: value for 'group' must be String literal; found extra Ident { sym: range, span: #0 bytes(803..808) }
  --> src/lib.rs:36:14
   |
36 |     #[export(group = "g1765" range = (0.0, 10.0, or_greater))]
   |              ^^^^^
```

It works nicely with IDEs, albeit it handles only one error at once :thinking: 

![image](https://github.com/user-attachments/assets/e3a3a3a8-c675-4c83-86ab-2e43c288a451)